analysis_of_products_in_stores/Task_2_1.py

```diff
@@ -0,0 +1,142 @@
+import os
+import csv
+from itertools import groupby
+from collections import Counter
+from common_functions.work_with_document import save_file
+
+def get_data(filename):
+    """function returns a dictionary from the datafile"""
+
+    ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+    with open(os.path.join(ROOT_DIR, filename)) as file:
+        reader = csv.reader(file)
+        next(reader)
+        main_dict = [
+            {
+                "ITEM_ID": row[0],
+                "STORE_ID": row[1],
+                "PRICE": float(row[2]),
+                "APPROVED_BY": row[3],
+            }
+            for row in reader
+        ]
+    return main_dict
+
+def sort_dict(dict, field):
+    """function returns a sorted dictionary for a field which was given like function's parameter and this parameter"""
+
+    sorted_dict = sorted(dict, key=lambda position: position[field])
```

Comment:
included it in 'group_dict' function

`def group_dict(dict, field):
    """function returns a grouped dictionary"""
    sorted_dict = sorted(dict, key=lambda position: position[field])
    grouped_dict = {
        key: [group for group in groups]
        for key, groups
        in groupby(sorted_dict[0], key=lambda position: position[sorted_dict[1]])
    }
    return grouped_dict`

Reply:
just fyi: to insert code block we have to use triple apostrophe sign. Please, check [this](https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks#fenced-code-blocks)

Reply:
okay, I will try it. I used "insert code" button on the toolbar.

Reply:
It can be even simple:
```
def group_dict(dict, field):
    return groupby(
        sorted(dict, key=lambda position: position[field]),
        key=lambda position: position[field],
    )
```
or inlined at all.

In your case `key: [group for group in groups]` is not neede, because gorupby returns you `{key -> [items with the key]}` dict.